src/plugins/embeddable/common/types.ts

```diff
@@ -67,4 +67,9 @@ export type EmbeddableInput = {
    * Visualization filters used to narrow down results.
    */
   filters?: Filter[];
+
+  /**
+   * Search session id to group searches
+   */
+  searchSessionId?: string;
```

Comment:
lets make a separate handler for this `getSessionId(): string | undefined`  ... 

 I actually started looking into this last week, but then got distracted by some type cleanup: https://github.com/elastic/kibana/pull/80643

Reply:
👍 Rebuilding this pr on top of #80643